### DISCUSSION TITLE
Retained mode `PaintCallback`s within `egui`

### DISCUSSION THREAD

**Discussion by tgross35** [emilk/egui#3650] [Q&A] [answered]
Is it possible now or in theory to skip redrawing the contents of a custom paint callback?

I have an application where the `egui` load is pretty light, but one pane using an `egui_wgpu::CallbackTrait` is getting heavy for both CPU and GPU. This content really doesn't change much, so it would be nice if `egui_wgpu::CallbackTrait::prepare` could return an optional value and `egui` skips the redraw of that rectangle's contents if it gets `None`.

I have no clue whether this is even feasible. If not, would this be possible when using `wgpu` + `winit` directly and using `egui` on top for the menu?

**Comment by Wumpf** [comment, marked as answer]
(This question may be old but maybe answering is still useful for someone out there!)

Since egui has to redraw the entire screen and doesn't keep draw results for single rectangles around this isn't easily possible. Also, this would need an elaborate mechanism to determine when a redraw is necessary. However, you can implement this yourself by drawing to a texture that you keep around and only update when needed. Then inside the `CallbackTrait` you (always) composite that texture to egui
